include/remill/Arch/Context.h

```diff
@@ -38,15 +39,23 @@ class DecodingContext {
  public:
   using ContextMap = std::function<DecodingContext(uint64_t)>;
```

Comment:
Is it desirable to have some kind of enumeration tag as a second argument, e.g to advise on the meaning of the address? Is it possible to have a conditional branch to the fall-through address, where the taken path changes the decoding mode?

Reply:
I think I generally would want to know why something more like an API isn't desirable. Something like of like: you can't even ask remill for the branch taken pc. Instead, you can ask it for the branch taken target, and it will give you back variant that is one of a (decoding context, pc), an indirect target tag, or not a target tag.

Reply:
So my explanation for why this was important is anvill knows the indirect target of indirect jump and remill knows how to update the context given address

Reply:
The case of conditional direct branch to the same fallthrough address with a mode change is possible but I don't know why we would handle it tbh. That would imply intentionally setting up an arm/thumb polyglot at some address for that to work. That being said given that the function didn't turn out how I hoped (ie. we cant just pass the branch target to resolve mode switches since it has to be in terms of the value prior to the mask) the lambda probably isn't strictly required. Honestly, I am not sure that we want to keep instruction categories at all. 

Instruction categories often fall over, granted we need them as hints for interprocedural flows, but I'd rather just return constant successors, handles into llvm to determine when those successors are reached, and contexts for them. Part of me wants to say that indirect contexts should be resolved by interpreting the llvm

Reply:
ie. realistically what I want is remill is it returns a [(pc, context, Optional<reaching_llvm_exression>)] where either the program counter or the context can be non-constant, in which case remill gives back an expression in terms of some members of the cpu state, that if resolved could be evaluated to the next pc and next context.

Reply:
The short answer is because I misevaluated and thought it would be possible to provide an expression from address->context for an arbitrary indirect target.